service/controller/machine_deployment_drainer_resource_set.go

```diff
@@ -179,5 +179,6 @@ func machineDeploymentDrainerLabelMapFunc(cr metav1.Object) map[string]string {
 	return map[string]string{
 		label.Cluster:           key.ClusterID(cr),
 		label.MachineDeployment: key.MachineDeploymentID(cr),
+		label.OperatorVersion:   key.OperatorVersion(cr),
 	}
```

Comment:
Adding the operator version label here so that only operators actually creating the `DrainerConfig` CRs drain and cleanup their own CRs. The map is configured to be used for both `drainerinitializer` and `drainerfinalizer` handlers.

Reply:
Why would different operators reconcile objects for same cluster? When there's cluster and machine deployment IDs, I'd expect that this already pins it down to single instance? What if cluster gets upgraded and for unknown reason there was `DrainerConfig` for a given node pool - wouldn't the CR get then orphaned?

Reply:
We just create and list `DrainerConfig` CRs. So `8.2.0` would do the finalizing that `8.3.0` would do as well. Now with the additional operator version label the operators only consider their own managed CRs.

Reply:
Upgrade and scaling should not happen together anyway. When upgrade is started the new operator version is reconciling everything of the cluster. Combined with the deletion on timeout this should be fine.

Reply:
> Upgrade and scaling should not happen together anyway.

Is `cluster-autoscaler` actively disabled during upgrade?

Reply:
Not that I am aware of. The responsibility of that app just got recently into Firecracker and we did not have yet the chance to get around to it. Do you think this is necessary? How to do it?

Reply:
> Do you think this is necessary?

From abandoned CR point of you yes, but from real life usage no.

> How to do it?

No clue. Depends probably from many factors, but in general scaling the deployment to `0 replicas` would do it. Problem is though that then there's tight coupling between an app and provider operator.

Might be useful to check how others deal with it (if they even do). There could be even some labeling trick or something to signal CA that it should leave a given ASG alone until the label is removed.

Reply:
So it was a bug? Two operators would reconcile the same resource?

Reply:
I think so.